src/Microsoft.CodeQuality.Analyzers/CSharp/QualityGuidelines/CSharpRemoveEmptyFinalizersAnalyzer.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Linq;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeQuality.Analyzers.QualityGuidelines;
+
+namespace Microsoft.CodeQuality.CSharp.Analyzers.QualityGuidelines
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpRemoveEmptyFinalizersAnalyzer : AbstractRemoveEmptyFinalizersAnalyzer
+    {
+        protected override bool IsEmptyFinalizer(SyntaxNode methodBody, SymbolAnalysisContext analysisContext)
+        {
+            var destructorDeclaration = (DestructorDeclarationSyntax)methodBody;
```

Comment:
yes , we should revert this back to ioperation analyzer once the underline issue is fixed